aiida_quantumespresso/parsers/pw.py

```diff
@@ -103,16 +103,16 @@ def parse(self, **kwargs):
             logs_xml.pop('error')
         self.emit_logs(logs_stdout, logs_xml)
 
-        # If the both stdout and xml exit codes are set, there was a basic problem with both output files and there
-        # is no need to investigate any further.
-        if self.exit_code_stdout and self.exit_code_xml:
-            return self.exit(self.exit_codes.ERROR_OUTPUT_FILES)
-
         # First check for specific known problems that can cause a pre-mature termination of the calculation
         exit_code = self.validate_premature_exit(logs_stdout)
         if exit_code:
             return self.exit(exit_code)
 
+        # If the both stdout and xml exit codes are set, there was a basic problem with both output files and there
+        # is no need to investigate any further.
+        if self.exit_code_stdout and self.exit_code_xml:
+            return self.exit(self.exit_codes.ERROR_OUTPUT_FILES)
+
```

Comment:
Is this change necessary for your new exit code to work correctly? I remember there being a problem with the order in evaluation, but I would rather deal with this then in separate PR first in order for it to not get lost in the rest of the changes in this PR.

Reply:
Yeah, the test will fail otherwise. Should I pull out 296c601d4cd722663012dff9f3e39ea64a9be63f into a separate PR? We could merge it ahead of this one.

Reply:
Yeah, since potentially affects the behavior quite a bit, I want to make sure what it does and it doesn't mess with my production runs

Reply:
Ok, see #439. I'll rebase this PR once that's done.

Reply:
PR is merged, rebase away!

Reply:
Done, ready for another review!